QATCH/processors/CurveOptimizer.py

```
@@ -890,19 +1071,41 @@ def correct_drop_effects(
+            # right expansion having extended the region past the spike peak into recovery
+            # (where the endpoint value can drop below the start value even for a genuine
+            # upward spike, causing the old endpoint comparison to wrongly skip the region).
+            region_diss_smooth = smooth_diss_full[region[0] : region[-1] + 1]
+            peak_diss = float(np.max(region_diss_smooth))
+            baseline_diss = min(float(region_diss_smooth[0]), float(region_diss_smooth[-1]))
+            full_range = float(np.max(smooth_diss_full) - np.min(smooth_diss_full))
+            if full_range > 0 and (peak_diss - baseline_diss) / full_range < 0.05:
                 Log.d(
                     self.TAG,
-                    f"Skipping region from {relative_time[region[0]]} to {relative_time[region[-1]]}",
+                    f"Skipping region {relative_time[region[0]]:.3f}–{relative_time[region[-1]]:.3f}: "
```

Ruff (0.15.20)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)